![https://github.com/GuanyuQiao98/dual-SCDR](https://img.shields.io/badge/version-1.0.0-blue)

# dual-SCDR: A Dual-Channel Foundation Model for Single-Cell Drug Response Prediction and Gene Co-expression Network Inference

This repository is the official implementation of dual-SCDR. 

<hr>



## ðŸ“– Brief Introduction

**dual-SCDR** provides a unified and interpretable framework for modeling drug response heterogeneity at single-cell resolution.  
It offers a foundation for more precise and mechanism-informed therapeutic strategies in **precision oncology**.

## ðŸ“¦ Requirements

Please install the following dependencies before running the code:

```
accelerate                0.27.0                  
deepspeed                 0.16.3               
dgl                       2.4.0.th21.cu118      
h5py                      3.12.1            
huggingface_hub           0.24.6         
igraph                    0.11.8                 
networkx                  3.3             
numpy                     1.26.4         
pandas                    2.2.2          
pillow                    11.0.0          
pyarrow                   16.1.0       
python                    3.10.15             
pytorch                   2.1.2           
pytorch-cuda              11.8                 
scanpy                    1.11.0                  
scikit-learn              1.5.2                    
scipy                     1.11.4               
tokenizers                0.20.3                
torch-geometric           2.6.1             
transformers              4.31.0 
```


## ðŸ“‚ Project Structure


This project mainly contains the following parts.
```
â”œâ”€â”€ utils/ # Utility functions and helper classes
â”‚ â””â”€â”€ utils.py # General utility functions
â”œâ”€â”€ dataprocess.py # Data preprocessing for dual-SCDR
```


## ðŸ“Š Dataset

The raw pharmacogenomic dataset can be obtained from the  
[DRMref Database](https://ccsm.uth.edu/DRMref/index.html).



## ðŸš€ Quick Start

1. **Download the dataset**  
   Get the raw data from the [DRMref Database](https://ccsm.uth.edu/DRMref/index.html).

2. **Convert to `h5ad` format**  
   Use tools like [Scanpy](https://scanpy.readthedocs.io/) to convert the downloaded data into `.h5ad` format.

3. **Run preprocessing**  
 >> python  dataprocess.py











